### PR TITLE
Day tab small by scroll

### LINKED ISSUE #103: Make day tab small by scroll
I would like to see cool animation for the scroll 🆒 
image

### PR DESCRIPTION
**Pull request by kako351**
## Issue
- close #103 

## Overview (Required)
- Day tab smaller by scroll.
- If the scroll position is below Top, the day tab will be smaller.
- Maintain day tab size when switching current page.
- Maintain day tab size when switching timetable to list.

Please point out any problems with implementation. 🙏 

## Links
-

## Screenshot
Before | After
:--: | :--:
<video src="https://user-images.githubusercontent.com/33022494/189489591-4d5ed9cb-5c4c-4dfc-99fa-5bfc5d66aa65.mp4" />| <video src="https://user-images.githubusercontent.com/33022494/189489606-a38db512-30ac-4e1c-8aaa-277636bfcade.mp4" />

### CODE REVIEW

**Comment by takahirom** [comment]
🆒

**Comment by takahirom** [comment]
Please fix conflict 🙏

**Comment by takahirom** [comment]
Let me see this tomorrow. 👀

**Comment by kako351** [comment]
Conflict resolved. 🙋 

> Let me see this tomorrow. 👀

Yes, thank you. 🙏

File: feature-sessions/src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/Timetable.kt [hunk 1]
```diff
@@ -359,6 +372,11 @@ class ScreenScrollState(
     val maxY: Float
         get() = _scrollY.lowerBound ?: 0f
 
+    val isScrollYProgress
+        get() = _isScrollYProgress.value || _scrollY.isRunning
+    val isScrollXProgress
+        get() = _isScrollXProgress.value || _scrollX.isRunning
+
```
Comment (kako351) [inline feature-sessions/src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/Timetable.kt:379]: Get the state during a drag or scroll animation.

- `_scrollY(Animatable).isRunning` returns false while dragging the screen
- Set `_isScrollYProgress` to true while dragging.

**Comment by mhidaka** [comment]
@kako351 Thanks for your quick work! now, we'are resolved conflict! I can't wait for tomorrow reveiw 😉

File: feature-sessions/src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/Sessions.kt [hunk 2]
```diff
@@ -458,3 +498,48 @@ fun SessionsLoadingPreview() {
...
+    val pagerState: PagerState,
+    val timetableStates: List<TimetableState>,
+    val sessionsListStates: List<LazyListState>
+) {
+    private var _scrollingPage = pagerState.currentPage
+
+    private val _isScrollTop = mutableStateOf(true)
+    val isScrollTop get() = _isScrollTop.value
+
+    suspend fun scrollTimetable() {
```
Comment (takahirom) [inline feature-sessions/src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/Sessions.kt:524]: I don't think we need suspend modifier here 👀

**Comment by takahirom** [comment]
@kako351 I pushed some changes to use derivedStateOf instead of LaunchedEffect. What do you think?

File: feature-sessions/src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/Sessions.kt (hunk 2 above)
Comment (kako351) [reply feature-sessions/src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/Sessions.kt]: Oh I see... That's right！

**Comment by kako351** [comment]
@takahirom  I confirmed pushed commit. Awesome code ! But didn't keep day tab size when switching current page.
Do you mean... was it not necessary to maintain the tab size when switching current page?

**Comment by takahirom** [comment]
I don't have a preference. But if there is no issue with it, I would like to keep it clean.

File: feature-sessions/src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/SessionDayTab.kt [hunk 3]
```diff
@@ -48,16 +57,22 @@ internal fun SessionDayTab(
...
-                text = "${5 + index}",
-                style = TextStyle(
-                    textAlign = TextAlign.Center,
-                    color = MaterialTheme.colorScheme.onSecondaryContainer,
-                    fontSize = 24.sp,
-                    fontWeight = FontWeight(500)
-                ),
-                modifier = Modifier.fillMaxWidth()
-            )
+            AnimatedVisibility(
```
Comment (takahirom) [inline feature-sessions/src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/SessionDayTab.kt:60]: Thanks for introducing AnimatedVisibility 👍

**Comment by kako351** [comment]
OK, I like your commited code. Your review and refactored code thank you.

**Comment by takahirom** [comment]
Certainly, I thought about adding other states, but it seems that an implementation similar to the original implementation is necessary to leave the tab state. It must have been hard to think about this. I didn't use it once, but I think it's a code that will certainly be a reference when realizing such specifications. Thank you very much. 🙏

**Comment by kako351** [comment]
Thank you your review and merged! It was difficult, but it was fun!

**Pull request merged**